src/custom/state/orders/updaters/ApiOrdersUpdater.ts

```diff
@@ -0,0 +1,188 @@
+  const outputToken = getTokenFromMapping(buyToken, chainId, allTokens)
+
+  const apiStatus = classifyOrder(order)
+  const status = statusMapping[apiStatus]
+
+  if (!status) {
+    console.warn(`ApiOrdersUpdater::Order ${id} in unknown internal state: ${apiStatus}`)
+    return
+  }
+  if (!inputToken || !outputToken) {
+    console.warn(
+      `ApiOrdersUpdater::Tokens not found for order ${id}: sellToken ${!inputToken ? sellToken : 'found'} - buyToken ${
+        !outputToken ? buyToken : 'found'
+      }`
+    )
+    return
+  }
+
+  const storeOrder: Order = {
+    ...order,
+    inputToken,
+    outputToken,
+    id,
+    creationTime,
+    summary: '',
+    status,
+    receiver,
+    apiAdditionalInfo: order,
+    isCancelling: apiStatus === 'pending' && order.invalidated, // already cancelled in the API, not yet in the UI
+  }
+  // The function to compute the summary needs the Order instance to exist already
+  // That's why it's not used before and an empty string is set instead
+  storeOrder.summary = computeOrderSummary({ orderFromStore: storeOrder, orderFromApi: order }) || ''
+
+  return storeOrder
+}
+
+function getMissingTokensAddresses(orders: OrderMetaData[], tokens: Record<string, Token>, chainId: ChainId): string[] {
+  const tokensToFetch = new Set<string>()
+
+  // Find out which tokens are not yet loaded in the UI
+  orders.forEach(({ sellToken, buyToken }) => {
+    if (!getTokenFromMapping(sellToken, chainId, tokens)) tokensToFetch.add(sellToken)
+    if (!getTokenFromMapping(buyToken, chainId, tokens)) tokensToFetch.add(buyToken)
+  })
+
+  return Array.from(tokensToFetch)
+}
+
+async function fetchTokens(
+  tokensToFetch: string[],
+  getToken: (address: string) => Promise<Token | null>
+): Promise<Record<string, Token | null>> {
+  if (tokensToFetch.length === 0) {
+    return {}
+  }
+
+  const promises = tokensToFetch.map((address) => getToken(address))
+  const settledPromises = await Promise.allSettled(promises)
+
+  return settledPromises.reduce<Record<string, Token | null>>((acc, promiseResult) => {
+    if (promiseResult.status === 'fulfilled' && promiseResult.value) {
+      acc[promiseResult.value.address] = promiseResult.value
+    }
+    return acc
+  }, {})
+}
+
+function filterOrders(orders: OrderMetaData[], tokens: Record<string, Token | null>, chainId: ChainId): Order[] {
+  return orders.reduce<Order[]>((acc, order) => {
+    const storeOrder = transformApiOrderToStoreOrder(order, chainId, tokens)
+    if (storeOrder) {
+      acc.push(storeOrder)
+    }
+    return acc
+  }, [])
+}
+
+export function ApiOrdersUpdater(): null {
+  const { account, chainId } = useActiveWeb3React()
+  const allTokens = useAllTokens()
+  const tokensAreLoaded = useMemo(() => Object.keys(allTokens).length > 0, [allTokens])
+  const addOrUpdateOrders = useAddOrUpdateOrders()
+  const getToken = useTokenLazy()
+  const apiOrders = useApiOrders(account)
```

Comment:
nit: the name `apiOrders` would make me think this is an API for getting orders, but is just orders from an API
Also, calling our endpoints just API can be confusing. Maybe we give another name?